subsys/console/getchar.c

```diff
@@ -19,9 +19,9 @@
 #error CONFIG_CONSOLE_PUTCHAR_BUFSIZE must be power of 2
 #endif
 
-static K_SEM_DEFINE(uart_sem, 0, UINT_MAX);
-static u8_t uart_ringbuf[CONFIG_CONSOLE_GETCHAR_BUFSIZE];
-static u8_t i_get, i_put;
+static K_SEM_DEFINE(rx_sem, 0, UINT_MAX);
+static u8_t rx_ringbuf[CONFIG_CONSOLE_GETCHAR_BUFSIZE];
```

Comment:
Naming changes OK with me.

I'd like to understand why we have a separate ring buffer implementation rather than using include/ring_buffer.h which ought to be completely suitable for this use-case.

Reply:
> I think I see why, sys_ring_buf_* works with data in 32-bit chunks

Exactly. As ringbuf is pretty simple data structure, it seemed easier to implement uint8_t ringbuf, rather than pack/unpack chars into 32-bit Zephyr's sys_ring_buf.